### DISCUSSION TITLE
Permission referencing relation field

### DISCUSSION THREAD

**Discussion by sapradhan** [directus/directus#14747] [Archive]
I have these models -

menuItem 
name - String 
datefor - Date 

userOrder
user - Directus user (many to one)
item - MenuItem foreign key (many to one)

```
POST /items/userOrder
			
{
	"user": "c498a5b4-31e3-4c5d-ab3e-2244a99ff30b",
	"item": 7
}
```

This POST request inserts a row in UserOrder collection as expected.

I want a constraint that new userOrder can only be created for current user with menuItems that have a future date as dateFor. 
So created a custom permission for create operation with field validation -  
```
{
	"_and": [
		{
			"user": {
				"_eq": "$CURRENT_USER"
			}
		},
		{
			"item": {
				"forDate": {
					"_gte": "$NOW"
				}
			}
		}
	]
}
```
![image](https://user-images.githubusercontent.com/5070211/181818407-6bfa6197-101b-4a3a-a21b-e10b420727f6.png)

My assumption was menuItem with given id is fetched, its dateFor is validated, but I get this error when I POST the same request. 

```
{
	"errors": [
		{
			"message": "\"item\" must be of type object",
			"extensions": {
				"code": "FAILED_VALIDATION",
				"field": "item"
			}[132238_6_20760331.pdf](https://github.com/directus/directus/files/9222307/132238_6_20760331.pdf)

**Comment by rijkvanzanten** [comment]
> My assumption was menuItem with given id is fetched, its dateFor is validated,

This assumption is wrong I'm afraid. Currently, Directus validates against the provided payload. If you provide an ID like `15`, it won't pass validation as you're requiring an object with `for_date >= $NOW`

**Reply by sapradhan** [reply]
Is there a way to support this use case ?

**Reply by rijkvanzanten** [reply]
Not yet, it's a new feature that has to be implemented in the platform

**Reply by sapradhan** [reply]
Thank you for your amazing work on Directus. Please add this to your feature backlog
